kolibri/plugins/qti_viewer/assets/src/views/AssessmentItemRef.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+
+  <AssessmentItem
+    v-if="assessmentItem"
+    :json="assessmentItem"
+  />
+
+</template>
+
+
+<script>
+
+  import jsonMixin from '../mixins/jsonMixin';
+  import qtiMixin from '../mixins/qtiMixin';
+  import xmlParse from '../utils/xml';
+  import AssessmentItem from './AssessmentItem';
+
+  const domParser = new DOMParser();
+
+  export default {
+    name: 'AssessmentItemRef',
+    components: {
+      AssessmentItem,
+    },
+    mixins: [jsonMixin, qtiMixin],
+    provide() {
+      return {
+        getDom: this.getAssessmentItemDom,
+      };
+    },
+    data() {
+      return {
+        assessmentItem: null,
+      };
+    },
+    computed: {
+      href() {
+        const relativeHref = this.json && this.json['@href'];
+        if (relativeHref) {
+          return new URL(
+            relativeHref,
+            new URL(this.getFilePath(), 'http://b.b/')
+          ).pathname.substring(1);
```

Comment:
A little clarification

```suggestion
          // iterative URL path construction using dummy domain
          return new URL(
            relativeHref,
            new URL(this.getFilePath(), 'http://example.com/')
          ).pathname.substring(1);
```

Reply:
I deliberately used a non-valid TLD to avoid any confusion, but yes.